bus-mapping/src/circuit_input_builder.rs

```diff
@@ -377,11 +458,53 @@ impl<'a> CircuitInputStateRef<'a> {
             .map(|s| s.stack.last().unwrap_or_else(|_| Word::zero()))
             .unwrap_or_else(Word::zero);
 
-        // Return from a call with a failure, without calling RETURN
-        if step.op != OpcodeId::RETURN
-            && step.depth != next_depth
-            && next_result == Word::zero()
-        {
+        // Return from a call with a failure
+        if step.depth != next_depth && next_result == Word::zero() {
+            if step.op != OpcodeId::RETURN {
+                // Without calling RETURN
+                return Some(match step.op {
+                    OpcodeId::REVERT => ExecError::ExecutionReverted,
+                    OpcodeId::JUMP | OpcodeId::JUMPI => ExecError::InvalidJump,
+                    OpcodeId::RETURNDATACOPY => {
+                        ExecError::ReturnDataOutOfBounds
+                    }
+                    _ => {
+                        panic!(
+                            "Cannot figure out call failure error in {:?}",
+                            step
+                        )
+                    }
+                });
+            } else {
+                // Calling RETURN
+                let call_ctx = self.tx_ctx.call_ctx();
+
+                // Return from a {CREATE, CREATE2} with a failure, via RETURN
+                if call_ctx.call != CallType::Root && call_ctx.is_create {
+                    let offset = step.stack.nth_last(0).unwrap();
+                    let length = step.stack.nth_last(1).unwrap();
+                    if length > Word::from(0x6000) {
+                        return Some(ExecError::MaxCodeSizeExceeded);
+                    } else if length > Word::zero()
+                        && !step.memory.0.is_empty()
+                        && step.memory.0.get(offset.low_u64() as usize)
+                            == Some(&0xef)
+                    {
+                        return Some(ExecError::InvalidCode);
+                    } else if Word::from(200) * length > Word::from(step.gas.0)
+                    {
+                        return Some(ExecError::CodeStoreOutOfGas);
+                    } else {
+                        panic!("Cannot figure out RETURN error in {:?}", step);
+                    }
+                } else {
+                    panic!("Cannot figure out RETURN error in {:?}", step);
+                }
+            }
+        }
+
+        // Return from a call with a failure, but error case not detected
+        if step.depth != next_depth && next_result == Word::zero() {
```

Comment:
Same here. We don't have any `else` even if we should not expect anything that doesn't follow the conditions exposed before.

Reply:
This was actually repeated code!  I've removed it.